web/services/checks.go

```diff
@@ -55,12 +56,17 @@ type ChecksService interface {
 }
 
 type checksService struct {
-	araService ara.AraService
-	db         *gorm.DB
+	db               *gorm.DB
+	araService       ara.AraService
+	premiumDetection PremiumDetectionService
 }
 
-func NewChecksService(araService ara.AraService, db *gorm.DB) *checksService {
-	return &checksService{araService: araService, db: db}
+func NewChecksService(db *gorm.DB, araService ara.AraService, premiumDetection PremiumDetectionService) *checksService {
+	return &checksService{
+		db:               db,
+		araService:       araService,
+		premiumDetection: premiumDetection,
```

Comment:
I would rename it to `premiumDetectionService`

Reply:
done